lib/modules/manager/gradle/extract/consistent-versions-plugin.ts

```diff
@@ -0,0 +1,187 @@
+ *
+ * @param availableFiles list of gradle build files found in project
+ * @param fileContents map with file contents of all files
+ */
+export function usesGcv(
+  availableFiles: string[],
+  fileContents: Record<string, string | null>
+): boolean {
+  if (
+    fileContents['build.gradle']?.includes(
+      'com.palantir.consistent-versions'
+    ) ||
+    fileContents['build.kts']?.includes('com.palantir.consistent-versions')
+  ) {
+    if (
+      availableFiles.includes(VERSIONS_PROPS) &&
+      availableFiles.includes(VERSIONS_LOCK)
+    ) {
+      logger.debug('This repo uses gradle-consistent-versions');
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * Confirms whether the provided file name is one of the two GCV files
+ */
+export function isGcvPropsFile(fileName: string): boolean {
+  return fileName === VERSIONS_PROPS || fileName === VERSIONS_LOCK;
+}
+
+/**
+ * Parses Gradle-Consistent-Versions files to figure out what dependencies, versions
+ * and groups they contain. The parsing goes like this:
+ * - Parse `versions.props` into deps (or groups) and versions, remembering file offsets
+ * - Parse `versions.lock` into deps and lock-versions
+ * - For each exact dep in props file, lookup the lock-version from lock file
+ * - For each group/regex dep in props file, lookup the set of exact deps and versions in lock file
+ *
+ * @param propsFileContent text content of `versions.props
+ * @param lockFileContent text content of `versions.lock`
+ */
+export function parseGcv(
+  propsFileContent: string,
+  lockFileContent: string
+): PackageDependency<GradleManagerData>[] {
+  const lockFileMap = parseLockFile(lockFileContent);
+  const [propsFileExactMap, propsFileRegexMap] =
+    parsePropsFile(propsFileContent);
+
+  const extractedDeps: PackageDependency<GradleManagerData>[] = [];
+
+  // For each exact dep in props file
+  for (const [propDep, versionAndPosition] of propsFileExactMap) {
+    if (lockFileMap.has(propDep)) {
+      const newDep: Record<string, any> = {
+        managerData: {
+          packageFile: VERSIONS_PROPS,
+          fileReplacePosition: versionAndPosition.filePos,
+        },
+        packageName: propDep,
+        currentValue: versionAndPosition.version,
+        currentVersion: versionAndPosition.version,
+        lockedVersion: lockFileMap.get(propDep),
+      } as PackageDependency<GradleManagerData>;
+      extractedDeps.push(newDep);
+    }
+  }
+
+  // For each regular expression (group) dep in props file
+  for (const [propDepRegEx, propVerAndPos] of propsFileRegexMap) {
+    for (const [exactDep, exactVer] of lockFileMap) {
+      if (propDepRegEx.test(exactDep)) {
+        const newDep: Record<string, any> = {
+          managerData: {
+            packageFile: VERSIONS_PROPS,
+            fileReplacePosition: propVerAndPos.filePos,
+          },
+          depName: exactDep,
+          currentValue: propVerAndPos.version,
+          currentVersion: propVerAndPos.version,
```

Comment:
How is this used / is there a reason you set `currentValue` and `currentVersion` to the same value?

Reply:
No, it's just me experimenting in lack of clear docs :) 
Which one is the correct one to use?

Reply:
So I looked around a bit, and found that `currentValue` is the original string parsed, while `currentVersion` is a value resolved by renovate - thus I now register `currentValue` only, see last commit.